co-log/example/1-intro/Intro.md

```diff
@@ -0,0 +1,109 @@
+# Intro: Using `LogAction`
+
+This tutorial is an introduction to `co-log`. It contains basic examples of
+using core data types and functions.
+
+You can run this tutorial by calling the following command:
+
+```shell
+cabal new-run intro
+```
+
+## Preamble: imports and language extensions
+
+Since this is a literate haskell file, we need to specify all our language
+extensions and imports up front.
+
+```haskell
+import Colog.Core (LogAction (..), (<&), logStringStdout)
+```
+
+## Core data type
+
+`co-log` is based on the following data type:
+
+```idris
+newtype LogAction m msg = LogAction
+    { unLogAction :: msg -> m ()
+    }
+```
+
+Logging action is a function from some message of user-defined type `msg` that
+performs all logic inside some monad `m`. In the `co-log` library **logger** is represented as a
+value. With such approach, you can modify the way you do logging by simply performing some
+transformations with the value you have.
+
+Let's first look at a very basic example of using just `putStrLn` for logging:
+
+```haskell
+example0 :: IO ()
+example0 = do
+    putStrLn "Example 0: First message"
+    putStrLn "Example 0: Second message"
+```
+
+Using `putStrLn` for logging is a very simple and basic approach for logging.
+When your application becomes bigger and more complex, you might want to bring
+some logging library into it.
+
+Now let's look at how you can `LogAction` instead of `putStrLn` to achieve the
+same goal. With `co-log` you need to have value of type `LogAction` that defines
+how you are going to do logging. So you configure your logging separately and
+then pass and use this `LogAction` value. See the following example for more
+details:
```

Comment:
Probably, it's better to point how this function is almost the same as the example0 and say something about how `logger` could look like, or say that you will cover it later? I guess it could help understand all together better, or how do you think?

Reply:
I think it's a good idea to say how the logging can look like, so readers can have better understanding. Thanks for the suggestion!